api_call_options.go

```diff
@@ -94,26 +98,42 @@ func (g *GetOptions) GetPage() int { // guaranteed int
 	return g.Page
 }
 
+func (g *GetOptions) GetCopy() *GetOptions {
+	if g == nil {
+		return &GetOptions{}
+	}
+	ret := *g
+	return &ret
+}
+
 // Including ensures that the variadic refs are included in a copy of the
 // options, resulting in expansion of the the referred sub-resources. Unknown
 // values within refs will be silently ignore by the API.
 func (g *GetOptions) Including(refs ...string) *GetOptions {
-	if g == nil {
-		return &GetOptions{Includes: refs}
-	}
-	out := *g
+	ret := g.GetCopy()
 	for _, v := range refs {
-		if !contains(out.Includes, v) {
-			out.Includes = append(out.Includes, v)
+		if !contains(ret.Includes, v) {
+			ret.Includes = append(ret.Includes, v)
 		}
 	}
-	return &out
+	return ret
+}
+
+func stripQuery(inURL string) string {
+	u, _ := url.Parse(inURL)
+	u.RawQuery = ""
+	return u.String()
 }
 
 // nextPage is common and extracted from all List functions
 func nextPage(meta meta, opts *GetOptions) (path string) {
```

Comment:
Would we be bonkers to implement `nextPage` as a channel or `go func` that would allow for all of the pages to be requested in parallel and reassembled in the expected order? We'd likely have to manage the queue, perhaps our HTTP libraries already offer features for this.

We can always revisit that later, it was just a thought.

Reply:
This a good idea, but maybe I'd leave it for another PR.